### ISSUE TITLE
Possible to link a SVG library instead of a single svg icon

### ISSUE THREAD

**Issue by pelord** [geostyler/geostyler#2404]
### Question

Is it possible, through IconLibrary 
https://github.com/geostyler/geostyler/blob/main/src/Component/Symbolizer/IconSelector/IconSelector.tsx#L45

to use a complete library instead of using a single icon (svg)?

ex:

```
const iconLibraries = [{
  name: 'Google Font Icons',
  icons: [{
    src: '....../path/to/the/whole/library.svg',
  }]
}, [{
  name: 'Material design icons',
  icons: [{
    src: 'https://raw.githubusercontent.com/Templarian/MaterialDesign-Angular-Material/master/mdi.svg'
  }]
}];
```

**Comment by alecarn** [comment]
Another question, is it possible to use the [typographic ligature](https://fonts.google.com/knowledge/glossary/ligature) to render icon?
By example, if we use a set of icons via Google fonts `<link href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined" rel="stylesheet" />` and we got all the icons names, is it something possible to render on the map?

**Comment by jansule** [comment]
> Is it possible, through IconLibrary
> https://github.com/geostyler/geostyler/blob/main/src/Component/Symbolizer/IconSelector/IconSelector.tsx#L45
>
> to use a complete library instead of using a single icon (svg)?

Currently this is not possible, but this might actually be a useful feature. This is also slightly related to https://github.com/geostyler/geostyler/issues/2390 where we are thinking of adding a handling for spritesheets.

If you are planning on providing a PR for this, this would get a :+1: from my side.

> Another question, is it possible to use the [typographic ligature](https://fonts.google.com/knowledge/glossary/ligature) to render icon?

This should theoretically be possible via the `<TextEditor>`, as long as the font is installed in the browser and the `<FontPicker>` gets passed the related font name. Then, by adding the character for the related icon to the input field, the icon should be visible. Unfortunately, it is currently not possible to pass through fonts. I just created a ticket for this: https://github.com/geostyler/geostyler/issues/2405.

The downside of this approach, even if it is working, is that users need to know the actual character that represents the icon, as there would be no visual hints.

A PR implementing an approach for the iconLibrary to display font icons would also get a :+1: from my side.

**Comment by alecarn** [comment]
> The downside of this approach, even if it is working, is that users need to know the actual character that represents the icon, as there would be no visual hints.

This depends on the implementation, for example if we have access to the name registers for the ligature feature of a specific icons set, we could have an IconSelector and the name logic would be hidden from an user perspective.

We will give a try to the TextEditor, thanks for you answer

**Comment by jansule** [comment]
Yes, there is still interest in this feature, but currently a lack of funding to work on it. Are you planning on implementing this feature?

**Comment by pelord** [comment]
This could be interesting, but I'm too new to your library at the moment.

In the meantime, could you please tell me a way to add icons manually?

**Comment by jansule** [comment]
You can take a look at the second example in https://geostyler.github.io/geostyler/latest/index.html#/Components/CardStyle/CardStyle, there you see how to include an icon library manually
